operator/cmd/mesh/manifest-migrate.go

```diff
@@ -83,14 +86,14 @@ func migrateFromFiles(rootArgs *rootArgs, args []string, l *Logger) error {
 		l.logAndPrint("no valid value.yaml file specified")
 		return nil
 	}
-	return translateFunc([]byte(value), l)
+	return translateFunc([]byte(value), mmArgs.force, l)
 }
 
 // translateFunc translates the input values and output the result
-func translateFunc(values []byte, l *Logger) error {
+func translateFunc(values []byte, force bool, l *Logger) error {
 	// First, try to translate from IstioControlPlane format.
 	icp := &icpv1alpha2.IstioControlPlane{}
-	if err := util.UnmarshalWithJSONPB(string(values), icp); err == nil {
+	if err := util.UnmarshalWithJSONPB(string(values), icp, false); err == nil {
```

Comment:
this may be insufficient of a test. i added a function in https://github.com/istio/istio/pull/20817 that tests the metadata in ICP to ensure it has the correct kind.